timdex_dataset_api/metadata.py

```diff
@@ -149,112 +127,13 @@ def create_metadata_structure(self) -> None:
                 exist_ok=True,
             )
 
-    def configure_duckdb_connection(self, conn: DuckDBPyConnection) -> None:
```

Comment:
Commenting here somewhat arbitrarily: all this DuckDB connection creation + configuration is moved to the new factory class.

Reply:
Again, a very smart refactor decision!